Automattic/jetpack
# Use DEWP's new auto-wp-polyfill support

## Proposed changes:

`@wordpress/dependency-extraction-webpack-plugin` has added the ability to detect a magic `/* wp:polyfill */` comment and only add `wp-polyfill` as a dependency if that is found. Since `wp-polyfill` is just a custom build of `core-js`, the intention is that `babel-plugin-polyfill-corejs3` (maybe via `@babel/preset-env`) would be used and then a custom Babel plugin would replace the `core-js` imports with the magic comment.

This updates our Babel config to do just that, and removes the blanket addition of `wp-polyfill` via `injectPolyfill` or otherwise.

### Effects

Removes unnecessary wp-polyfill from the following:
* packages/explat
* packages/forms block view.js
* packages/jitm
* packages/plugin-deactivation
* All Jetpack block view.js except Map, Recurring Payments, Subscriptions, Story, Nextdoor, and Podcast Player.
* packages/videopress `divi-editor/index.js`, `player-bridge.js`, and block view.js

Also it adds wp-polyfill as a dependency for a few things that were missing it before:
* Boost's `inline-liar.js`
* packages/connection scripts
* packages/forms `jetpack-forms-dashboard.js`
* Lots of mu-wpcom scripts:
  * `block-inserter-modifications.js`
  * `newspack-blocks-blog-posts-editor.js`
  * `newspack-blocks-blog-posts-view.js`
  * `newspack-blocks-carousel-editor.js`
  * `tags-education.js`
  * `verbum-comments.js`
  * `wpcom-block-description-links.js`
  * `wpcom-block-editor-nux.js`
  * `wpcom-blocks-event-countdown-editor.js`
  * `wpcom-blocks-timeline-editor.js`
  * `wpcom-global-styles-editor.js`
* js-packages/react-data-sync-client
* js-packages/svelte-data-sync-client

The following specifically weren't changed:
* packages/search and packages/wordads still unconditionally include the dep, and also embed core-js directly.

### Other information:


## Jetpack product discussion


Fixes #39452

## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Verify that using Jetpack blocks other than the ones mentioned doesn't trigger `wp-polyfill` on the frontend anymore.
* Other than that, 🤷. There are no real changes in the bundled code here, only the addition or removal of polyfills for older-but-still-supported browsers.

P.S. If building locally to test, you may want to delete any stray `.cache/` directories first. Babel seems not to notice changes in the preset and so incorrectly reuses cache entries from older builds.
